androidtv/constants.py

```diff
@@ -28,20 +28,32 @@
 #: Determine whether the device is awake
 CMD_AWAKE = "dumpsys power | grep mWakefulness | grep -q Awake"
 
-#: Get the current app
-CMD_CURRENT_APP = "CURRENT_APP=$(dumpsys window windows | grep mCurrentFocus) && CURRENT_APP=${CURRENT_APP#*{* * } && CURRENT_APP=${CURRENT_APP%%/*} && echo $CURRENT_APP"
+#: Parse the current app from the dumpsys outputand store it in the variable ``CURRENT_APP``
+CMD_PARSE_CURRENT_APP = 'CURRENT_APP=${CURRENT_APP#*ActivityRecord{* * } CURRENT_APP=${CURRENT_APP#*{* * } && CURRENT_APP=${CURRENT_APP%%/*} && CURRENT_APP=${CURRENT_APP%\\}*}'
+
+#: Get the current app (but don't echo it)
+CMD_GET_CURRENT_APP = "CURRENT_APP=$(dumpsys window windows | grep -E 'mCurrentFocus|mFocusedApp') && " + CMD_PARSE_CURRENT_APP
+
+#: Get the current app and echo it
+CMD_CURRENT_APP = CMD_GET_CURRENT_APP + ' && echo $CURRENT_APP'
```

Comment:
Looking at these constants (names), I'm confused as it gives the false pretence that there are 2 getters, when in fact there is only 1.

Would it maybe make sense to have `CMD_GET_CURRENT_APP` and `CMD_ECHO_CURRENT_APP`?

Although renaming `CMD_CURRENT_APP` to anything else will introduce any regressions, hence the reason I previously named the definition of `CURRENT_APP` as `CMD_DEFINE_CURRENT_APP`.

Reply:
I also thought about renaming `CMD_CURRENT_APP` to `CMD_ECHO_CURRENT_APP` but I didn't want to have to change all occurrences (although it's really not a big deal to do so, just a `sed` command).  

Another option is to prefix the other two `CURRENT_APP` commands with an underscore. 

Maybe `CMD_DEFINE_CURRENT_APP_VARIABLE` is the most clear name? (And maybe prefix it with an underscore?)

Reply:
Yes, `CMD_DEFINE_CURRENT_APP_VARIABLE` would most certain be less confusing than `CMD_GET_CURRENT_APP`.

Might I suggest to also update the comments describing the commands, something along the following?
```python
#: Assign focused application identifier to ``CURRENT_APP`` variable
CMD_DEFINE_CURRENT_APP_VARIABLE = "CURRENT_APP=$(dumpsys window windows | grep -E 'mCurrentFocus|mFocusedApp') && " + CMD_PARSE_CURRENT_APP

#: Output identifier for current/focused application
CMD_CURRENT_APP = CMD_DEFINE_CURRENT_APP_VARIABLE + ' && echo $CURRENT_APP'

#: Assign focused application identifier to ``CURRENT_APP`` variable (for a Google TV device)
CMD_DEFINE_CURRENT_APP_VARIABLE_GOOGLE_TV = 'CURRENT_APP=$(dumpsys activity a . | grep mResumedActivity) && ' + CMD_PARSE_CURRENT_APP

#: Output identifier for current/focused application (for a Google TV device)
CMD_CURRENT_APP_GOOGLE_TV = CMD_DEFINE_CURRENT_APP_VARIABLE_GOOGLE_TV + ' && echo $CURRENT_APP'
```

I don't think underscore prefix is needed, we also do not prefix `CMD_MEDIA_SESSION_STATE`, which is referenced from other commands.

Reply:
OK, let's go with `CMD_DEFINE_CURRENT_APP_VARIABLE`. 

Your suggestions are good. Would you mind submitting them, either as a pull request to this branch or as "suggested change" comments?

Reply:
Created a pull request (#237), could not comment a suggestion due to the line deletion in the diff.